### PR TITLE
 Add socket support for mysql data source

### LINKED ISSUE #12342: Support MySQL data source to connect via socket
It seems MySQL provisionning data source doesn't support MySQL socket (but configuring Grafana database as MySQL support it... even if not mentionned in documentation, if the host begin by / it switch to unix socket connection).

Actually i cannot use Grafana because only sockets are used in our infrastructure (security and performance need).

I try with this notation too, but with no luck:
database

Is it a way to use mysql socket for provisionning data source ? If yes it should be documented, if no could it be added please ?

At least documentation here http://docs.grafana.org/installation/configuration/#database should mention unix socket are supported and how to use it (see source code here please: https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/sqlstore.go#L137)

Thanks.

### PR DESCRIPTION
**Pull request by Yukinoshita-Yukino**
Currently mysql data source only supports tcp protocol as mentioned in #12342.
Add support for socket protocol.

Fixes #12342

### CODE REVIEW

File: pkg/tsdb/mysql/mysql.go [hunk 1]
```diff
@@ -20,10 +21,16 @@ func init() {
 func newMysqlQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
 	logger := log.New("tsdb.mysql")
 
+	var protocol string
```
Comment (marefr) [inline pkg/tsdb/mysql/mysql.go]: Please simplify this by initializing protocol to `tcp` and skip the else clause:

```go
protocol := "tcp"
if strings.HasPrefix(datasource.Url, "/") {
	protocol = "unix"
}
```

**Review by marefr — CHANGES_REQUESTED** [review]
Thanks! 

Looks really good. Just have a minor change request, see comment.

**Comment by Yukinoshita-Yukino** [comment]
Sure, done that.

**Review by marefr — APPROVED** [review]
Excellent :+1:

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Pull request merged**